atomic_reactor/plugins/exit_remove_worker_metadata.py

```diff
@@ -0,0 +1,38 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from atomic_reactor.plugin import ExitPlugin
+from atomic_reactor.constants import PLUGIN_REMOVE_WORKER_METADATA_KEY
+
+__all__ = ('RemoveWorkerMetadataPlugin', )
+
+
+def defer_removal(workflow, cf_map, osbs):
+    key = RemoveWorkerMetadataPlugin.key
+    workflow.plugin_workspace.setdefault(key, {})
+    workspace = workflow.plugin_workspace[key]
+    workspace.setdefault('cf_maps_to_remove', set())
+    workspace['cf_maps_to_remove'].add((cf_map, osbs))
```

Comment:
There is no need to store `osbs` here - it would be the same for all configmaps. Its easier to fetch `osbs` in `def run` once

Reply:
That's not true. Each configmap comes from a different cluster, requiring a different osbs api object.

Reply:
Nack on that. We store the osbs object from `orchestrate_build` exactly for this reason. Any plugin that needs to reach out to worker clusters will use it, just like `fetch_metadata_worker` does.
Spawning a new osbs instance is not as straight forward as you may think, as it requires reading the reactor config and client config again.

Reply:
Hmm, okay, lets leave it for further improvements then - I assumes its nearly effortless